force-app/main/default/classes/HandlerClass.cls

```diff
@@ -0,0 +1,80 @@
+/**
+ * File:        HandlerClass
+ * Project:     VsExerciseProjects
+ * Date:        July 28, 2021
+ * Created By:  Dragos Olaru
+ * *************************************************************************
+ * Description:  Class with handlers for the triggers
+ * *************************************************************************
+ * History:
+ * Date:       			            Modified By:  		           Description:
+ *
+ */
+
+public with sharing class HandlerClass {
+  /**
+   * @author Dragos Olaru
+   * @Date July 28, 2021
+   * @name BeforeAction
+   * @description Blocks setting another primary contact
+   * @return void
+   */
+
+  public void BeforeAction(Contact[] newContacts) {
+    for (Contact c : newContacts) {
+      if (c.Is_Primary_Contact__c) {
+        try {
+          Integer primaryContactCount = [
+            SELECT COUNT()
+            FROM Contact
+            WHERE
+              AccountId = :c.AccountId
+              AND Id != :c.Id
+              AND Is_Primary_Contact__c = TRUE
+          ];
+          if (primaryContactCount > 0) {
+            c.addError(Label.PrimaryContactAlert);
+          }
+        } catch (Exception e) {
+          System.debug('Error' + e.getMessage());
+        }
+      }
+    }
+  }
+
+  /**
+   * @author Dragos Olaru
+   * @Date July 28, 2021
+   * @name AfterAction
+   * @description After insert trigger handle
+   * @return void
+   */
+
+  public void AfterAction(Contact[] newContacts) {
+    for (Contact c : newContacts)
+      if (c.Is_Primary_Contact__c) {
+        Database.executeBatch(new BatchableClass(c.AccountId));
```

Comment:
this is going to hit another limitation (flexqueue) if there is more than 100 contacts processed at the same time. ideally batch class should have only ones and it will process all contacts